integrationTests/proxyWithChainSimulator.go

```diff
@@ -226,26 +226,45 @@ func (instance *proxyWithChainSimulator) DeploySC(ctx context.Context, wasmFileP
 
 // GetTransactionResult tries to get a transaction result. It may wait a few blocks
 func (instance *proxyWithChainSimulator) GetTransactionResult(ctx context.Context, hash string) (data.TransactionOnNetwork, error) {
-	txResult, errGet := instance.proxyInstance.GetTransactionInfoWithResults(ctx, hash)
-	if errGet == nil && txResult.Data.Transaction.Status == transaction.TxStatusSuccess.String() {
-		return txResult.Data.Transaction, nil
+	txResult, err := instance.getTxInfoWithResultsIfTxProcessingFinished(ctx, hash)
+	if err == nil && txResult != nil {
+		return *txResult, nil
 	}
 
 	// wait for tx to be done, in order to get the contract address
-	timeoutTimer := time.NewTimer(instance.roundDuration * 5)
+	timeoutTimer := time.NewTimer(instance.roundDuration * 20)
```

Comment:
this timer is only a fallback protection as the `proxyInstance.ProcessTransactionStatus` should always give a  result.
However, the approach is a little bit misleading because it will block the tes forever if one transaction is failed. I will try to come up with a better solution to this.

Reply:
updated as suggested